### DISCUSSION TITLE
Next release -- Please test

### DISCUSSION THREAD

**Discussion by kebekus** [Akaflieg-Freiburg/enroute#316] [Team Posts]
Dear all,

The next release of **Enroute Flight Navigation** is coming up, and much has changed internally. I expect a longer-than-usual beta test phase. I would greatly appreciate your feedback if you could try the new features. I am particularly interested to learn if the app reliably recognizes the next waypoint when you follow a route (critical cases might include: routes with sharp angles, return routes, switching the app on and off during the flight, switching the app off before a waypoint, )

Best wishes,

Stefan.

-----

### ETE, ETA, and rubber band 

If you fly along a route, the app will show the remaining distance, ETE, and ETA for the next waypoint and the final waypoint. The moving map shows a red "rubber bad" between the current position and the next waypoint. The information is shown under the following conditions:

- The user has set a route.
- The user has set aircraft cruise speed and wind information.
- The aircraft is flying.
- The plane is closer than three nautical miles to the route.
- The distance to the final waypoint is more than three nautical miles.

### Behavior on direct-to

When double-tap into the moving map, the waypoint description dialog will open, and you have the choice to set a quick "direct to" route. The app will now show the rubber band to the waypoint when you do that. The following conditions must be met:

- The user has set aircraft cruise speed and wind information.
- The aircraft is flying.
- The distance to the final waypoint is more than three nautical miles.

### Map update notifications

The app now uses Android notifications to alert you when map updates are available. The alert will not show up while in flight.

**Comment by TTC70** [comment]
Hello Stefan,
How do I get the beta?
At the moment I believe I have the "market" version.
Cheers.
Antonio


Il Mar 15 Mar 2022, 07:26 Stefan Kebekus ***@***.***> ha
scritto:

**Comment by kebekus** [comment]
Thanks for asking. You can use this link to subscribe to become a beta tester 

- https://play.google.com/apps/testing/de.akaflieg_freiburg.enroute

It goes without saying that you can leave the beta-program anytime.

Best,

Stefan.

**Comment by TTC70** [comment]
Thanks Stefan,
at this point I am asking myself if becoming a beta tester might be useful
for you once I do not fly yet.
I am still a student pilot (ultralight) and I do not own an airplane, I was
thinking of using the app as "car navigator" to set the flight plan on the
routes I usually use by car...
Might it be useful for you and for testing the app?
Cheers
Antonio

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Mail
priva di virus. www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

Il giorno mar 15 mar 2022 alle ore 16:28 Stefan Kebekus <
***@***.***> ha scritto:

**Comment by kebekus** [comment]
> Might it be useful for you and for testing the app?

Sure.

**Comment by TTC70** [comment]
Great.... I will proceed to ask to be a beta tester then.


<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Mail
priva di virus. www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

Il giorno mer 16 mar 2022 alle ore 09:16 Stefan Kebekus <
***@***.***> ha scritto:

**Comment by Ivan081** [comment]
Of I set km/h as Speed and save the airplane app do not remember km/h

Il mer 16 mar 2022, 09:18 TTC70 ***@***.***> ha scritto:

**Comment by TTC70** [comment]
I confirm..... it seems the App does not save the new speed unit.
I tried to change the speed unit, save the airplane, close the app, reopen
it and the speed unit was still in mph.
I hope this helps
Cheers
Antonio

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Mail
priva di virus. www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

Il giorno mer 16 mar 2022 alle ore 13:15 Ivan ***@***.***> ha
scritto:

**Comment by kebekus** [comment]
@Ivan081 @TTC70 Thank you for your report. I will open an issue, you should see some activity in a few minutes.

**Comment by MichaelHorbaschk** [comment]
Hello Stefan,

I have tested the Beta yesterday during a passenger flight.
I noted, that the Switch to the next waypoint is done at 3 NM regardless
how you approach. Even If exactely on course the Switch ist at 3NM before
the WP.
Is this desired?
I think the ETE ETA based on set speed is dangerous as speed may be old and
not includes wind (especially for Destination).
The aircraft speed does not accept values bigger then 399 kts. (A320 was
faster)
The True Track may be a more helpful info then 2 times (ETE ETA)
I may add a chapter "Navigation along a route" to the handbook next
weekend. I would request you to create the reference for a dynamic
Screenshot showing the new headlines and the Rubberband later. For the
start IT will work without the Illustration.

Viele Grüße
Michael Horbaschk

Stefan Kebekus ***@***.***> schrieb am Di., 15. März 2022,
07:26:

**Comment by kebekus** [comment]
@MichaelHorbaschk Thank you for your report. It is a bug that the switch to the next waypoint is done at 3 NM regardless of how you approach it. I will fix the bug and upload a new version in the next few days.

**Comment by kebekus** [comment]
MichaelHorbaschk wrote: 

> I think the ETE ETA based on set speed is dangerous as speed may be old and not includes wind (especially for Destination).

The ETE and ETA computation should include wind. I will double-check that this is indeed the case. Thank you for the comment.

**Comment by kebekus** [comment]
MichaelHorbaschk wrote:

> The aircraft speed does not accept values bigger then 399 kts. (A320 was faster)

True. I fly a Diamond DA20. What would be a good upper speed limit?

**Comment by kebekus** [comment]
MichaelHorbaschk wrote:

> The True Track may be a more helpful info then 2 times (ETE ETA)

I am unsure, but I am open to changes. Let's discuss that when the bugs are fixed!

**Comment by kebekus** [comment]
MichaelHorbaschk wrote:

> I may add a chapter "Navigation along a route" to the handbook next weekend. I would request you to create the reference for a dynamic Screenshot showing the new headlines and the Rubberband later. For the start IT will work without the Illustration.

Thanks! This chapter already exists in Section 2; check [here](https://akaflieg-freiburg.github.io/enrouteText/manual/02-steps/plan.html). The illustrations exist in the "working" branch of the GIT repository (I have not merged them into the 'master' branch, because the functionality is not available to regular users yet). 

There is still a lot of room for improvement, and your contributions are highly welcome.

Best,

Stefan,

**Comment by kebekus** [comment]
@MichaelHorbaschk Program version 2.15.93 is now available for Beta-testers at Google play. I have hopefully been able to fix the issue. If you get a chance to try this version in the air, I would appreciate your feedback. No matter what: thank you for your help!

**Comment by TTC70** [comment]
Stefan,
I just finished a 1h10m flight with MSFS2020: LIRP - LIEA (flight plan GOVG5C GOVGO Z254 ELB Q125 ROMPO L5 MINKA MINK2Z.... I know this is an IFR flight plan, i reproduced the same waypoints on Enroute).
Here what I found out regarding ETA/ETE menu, waypoints and rubber band.
1. overall the ETA/ETE menu works.
2. overall the rubber band shows the direction to the next waypoint, but only when ETA/ETE menu is on the screen.
"Overall" means that:
1. sometimes the abovementioned menu, and subsequently also the rubber band, disappears (the app is connected to the sim, the plane is correctly tracked and moves quite seamlessly).
2. the switch to the "next waypoint ETA" and next rubber band target is not constant. Some waypoint is kept until 1,4 miles to it, some other until 0,7 miles, some other until 0,4 miles.
3. at the beginning of the flight neither the menu nor the rubber band appeared. they both appeared some time later (after the first waypoint if i am not mistaken).
4. at the end of the flight, at the third WP from the destination, both menu ETA/ETE and rubber band disappeared, reappearing close to the second WP from destination showing correctly the destination as next waypoint.
....
I hope this helps.

P.S.
I am not sure if this is a programmer choice: when the app is not connected to MSFS, the maps goes apparently on the equator off the coast of Africa. The current location appears only when the app is connected through fs2ff (just an idea... maybe a "Home" location could be setup?).

**Comment by kebekus** [comment]
Dear all,

thank you for your feedback! I understand from all the reports that many users are confused because the Remaining Route Infobox switches on and off during the flight, without apparent reason. Some users also requested that the Infobox is visible on the ground, before take-off.

I have improved the app accordingly and uploaded a new version (=2.15.94) to Google Play. Instead of switching the Infobox off, the app now shows a reason why the box is no longer visible. Also, the box is shown before take-off (except start point = end point, because then the app cannot tell if we are "before start" or "after landing"). If you have the chance, please test and send me your feedback!

Best wishes · Thank you for the help,

Stefan.

**Comment by kebekus** [comment]
@TTC70: Item #4: The app switches to the next leg of your route whenever you are closer than 3nm to the leg AND your direction of flight matches. Depending on True Track, the app may therefore switch earlier or later.

@TTC70: Item #5: That was a design decision, because I wanted to keep the display clear while you are in the traffic patter. However, I saw that users were confused, so I changed this behavior.

@TTC70: Item #6: Same as #5

@TTC80: ITem PS: The app uses the last known location. I would guess that MSFS sets the location somewhere near the equator on shutdown. As soon as you go outside and your device GPS works, you should come back to a reasonable place.

**Comment by TTC70** [comment]
Stefan, using the app more extensively, I have come to realize that it is more intuitive and of immediate understanding if the flight plan path is magenta.
Understanding that each programmer has his/her own way to produce software, maybe you can give the user the option to keep your original color of the flight plan path, or choose another color. Or at least between your original color and magenta.
Magenta is widely used by most of navigators (both for cars and airplanes) and maybe the user might be attracted to the app a little more if there is also this feature.

**Comment by kebekus** [comment]
@TTC70: will open a new issue report for your request

**Comment by MichaelHorbaschk** [comment]
Hallo Stefan,

habe gestern einen längeren Flug gemacht.
Alle Funktionen von Enroute haben wie erwartet funktioniert. Das Umschalten
der WP war erwartungsgemäß am Punkt oder Parallel. Auch das Wiederaufnehmen
der Route nach Abweichen hat funktioniert.
Die Anzeige eines Kurses im oberen Fenster wäre aus meiner Sicht sehr
hilfreich.
Die Uhrzeit am Punkt ist eher Überflüssig, da einfach im Kopf auszurechnen.

Viele Grüße

Michael Horbaschk

Stefan Kebekus ***@***.***> schrieb am Mo., 28. März 2022,
07:16:

**Comment by TTC70** [comment]
Hello all,
I carefully read the comment below.
As new VFR pilot i find very useful the information regarding the ETA to
the next waypoint.
I believe it can help a lot and lighten the workload of the pilot.
Thanks
Cheers.
Antonio

Il Mar 11 Ott 2022, 08:55 MichaelHorbaschk ***@***.***> ha
scritto:

**Comment by kebekus** [comment]
@MichaelHorbaschk Thanks for your suggestion. Starting with the next version of the app, the (true) course to the next waypoint is now shown in the RemainingRouteBar at the top of the screen. The course is however only shown if the display is wide enough -- the slim design of many modern phones puts some limits on much I can show.

**Comment by Ivan081** [comment]
Hi I have a question from some pilots, Is possibile ti add notam info to
the app?



Il gio 29 dic 2022, 07:55 Stefan Kebekus ***@***.***> ha
scritto:

**Comment by kebekus** [comment]
@Ivan081 Yes, there is no technical problem with adding NOTAMS to the app. The **only** problem is that I am programming the app alone, so that development is slow. I have received literally thousands of requests for additional features, but so far not a single person was willing to help with the programming. 

If you know people who might be interested in joining the development, that would be more than welcome!

Best,

Stefan.

**Comment by kebekus** [comment]
@Ivan081 I added you to the Issue for "NOTAMS", so we can continue the discussion there.
